app/main.py

```
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for i in people:
+        person = Person(i["name"], i["age"])
+        result.append(person)
+
+    for i in people:
+        person = Person.people[i["name"]]
+        if i.get("wife"):
+            spouse = Person.people.get(i["wife"])
+            if spouse:
+                person.wife = spouse
+                spouse.husband = person
+        if i.get("husband"):
+            spouse = Person.people.get(i["husband"])
+            if spouse:
+                person.husband = spouse
+                spouse.wife = person
```

The code correctly sets spouse relationships if the spouse exists in the `people` dictionary. However, it assumes that the spouse names are always present in the dictionary. If a spouse name is not found, it will silently skip setting the relationship. Consider adding error handling or logging to address cases where a spouse is not found.